### PR TITLE
Windows native build via AppVeyor

### LINKED ISSUE #39: Windows native build
[Cygwin](https://www.cygwin.com/) seems to have everything we need to build pairinteraction.  This could then be integrated with [AppVeyor](https://www.appveyor.com/) to create native Windows builds.  This would save us the horrible cross compilation.

Another possibility is to build pairinteraction using Microsoft Visual Studio.

### PR DESCRIPTION
**Pull request by seweber**
Closes #39

### CODE REVIEW

File: calc/HamiltonianOne.cpp [hunk 1]
```diff
@@ -407,7 +407,7 @@ void HamiltonianOne::build() {
 #pragma omp parallel for schedule(static, 1)
 
     // Loop through steps
-    for (size_t step = 0; step < nSteps; ++step) {
+    for (int step = 0; step < static_cast<int>(nSteps); ++step) {
```
Comment (hmenke) [inline calc/HamiltonianOne.cpp]: No! `size_t step = 0; step < nSteps; ++step` was perfectly correct.

File: calc/HamiltonianTwo.cpp [hunk 2]
```diff
@@ -286,7 +286,7 @@ void HamiltonianTwo::calculate(const Configuration &conf_tot) {
     std::vector<bool> necessary_tmp(basis->size(), false);
 
 #pragma omp parallel for
-    for (size_t i = 0; i < nSteps_one; ++i) {
+    for (int i = 0; i < static_cast<int>(nSteps_one); ++i) {
```
Comment (hmenke) [inline calc/HamiltonianTwo.cpp]: Same here.

File: calc/HamiltonianTwo.cpp [hunk 3]
```diff
@@ -516,7 +516,7 @@ void HamiltonianTwo::calculate(const Configuration &conf_tot) {
         mat_single.resize(symmetries.size());
 
 #pragma omp parallel for
-        for (size_t idx_symmetry = 0; idx_symmetry < symmetries.size(); ++idx_symmetry) {
+        for (int idx_symmetry = 0; idx_symmetry < static_cast<int>(symmetries.size()); ++idx_symmetry) {
```
Comment (hmenke) [inline calc/HamiltonianTwo.cpp]: Same here.

File: calc/HamiltonianTwo.cpp [hunk 4]
```diff
@@ -536,8 +536,8 @@ void HamiltonianTwo::calculate(const Configuration &conf_tot) {
 
         mat_multipole_transformed.resize(symmetries.size()*(idx_multipole_max+1));
 
-#pragma omp parallel for collapse(2)
-        for (size_t idx_symmetry = 0; idx_symmetry < symmetries.size(); ++idx_symmetry) {
+#pragma omp parallel for
+        for (int idx_symmetry = 0; idx_symmetry < static_cast<int>(symmetries.size()); ++idx_symmetry) {
```
Comment (hmenke) [inline calc/HamiltonianTwo.cpp]: Same here.  Also, why was loop collapsing removed?

File: calc/HamiltonianTwo.cpp [hunk 5]
```diff
@@ -551,10 +551,10 @@ void HamiltonianTwo::calculate(const Configuration &conf_tot) {
 
     std::cout << ">>TOT" << std::setw(7) << nSteps_two*symmetries.size() << std::endl;
 
-#pragma omp parallel for collapse(2) schedule(static, 1)
+#pragma omp parallel for schedule(static, 1)
 
     // Loop through steps
-    for (size_t step_two = 0; step_two < nSteps_two; ++step_two) {
+    for (int step_two = 0; step_two < static_cast<int>(nSteps_two); ++step_two) {
```
Comment (hmenke) [inline calc/HamiltonianTwo.cpp]: Same here.

File: calc/HamiltonianTwo.cpp [hunk 6]
```diff
@@ -551,10 +551,10 @@ void HamiltonianTwo::calculate(const Configuration &conf_tot) {
 
     std::cout << ">>TOT" << std::setw(7) << nSteps_two*symmetries.size() << std::endl;
 
-#pragma omp parallel for collapse(2) schedule(static, 1)
+#pragma omp parallel for schedule(static, 1)
```
Comment (hmenke) [inline calc/HamiltonianTwo.cpp]: Why was collapsing removed?  Why is there the need for static scheduling?  If this is meant to impose some kind of ordering, don't parallelize altogether please.

File: gui/CMakeLists.txt [hunk 7]
```diff
@@ -69,12 +71,20 @@ if( WITH_DMG )
...
+      COMMAND ${PYINSTALLER} ${CMAKE_CURRENT_BINARY_DIR}/startgui.spec
+      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/pairinteraction
+    )
+  else(CMAKE_HOST_WIN32)
+	find_package(PythonInterp 3 REQUIRED)
+	add_custom_command(TARGET gui POST_BUILD
+      COMMAND ${PYTHON_EXECUTABLE} -m compileall -q ${CMAKE_CURRENT_BINARY_DIR}/pairinteraction
+      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/pairinteraction
+	)
+  endif(CMAKE_HOST_WIN32)
```
Comment (hmenke) [inline gui/CMakeLists.txt]: Indentation is weird.

File: calc/HamiltonianOne.cpp (hunk 1 above)
Comment (seweber) [inline calc/HamiltonianOne.cpp]: No, unfortunately, Visual Studio 2015 does fully support only the OpenMP 2.0 specifications.

File: calc/HamiltonianTwo.cpp (hunk 4 above)
Comment (seweber) [inline calc/HamiltonianTwo.cpp]: Both changes are required because of lacking support of Visual Studio 2015.

File: calc/HamiltonianOne.cpp (hunk 1 above)
Comment (hmenke) [inline calc/HamiltonianOne.cpp]: Could we then at least do
```
int nSteps_i = static_cast<int>(nSteps);
for (int step = 0; step < nSteps_i; ++step) {
```
to not have the runtime cost of a `static_cast` at every iteration or we just leave it as
```
for (int step = 0; step < nSteps; ++step) {
```
and tolerate the compiler warning.

File: calc/HamiltonianTwo.cpp (hunk 6 above)
Comment (seweber) [inline calc/HamiltonianTwo.cpp]: Same here. Concerning static scheduling, the ordering is requested to build the potential lines starting from one site of the plot window and ending up on the other site. This successive way of plotting allows to start the drawing of potential lines already before everything is calculated. In my opinion, your suggestion won't work here. We really need static parallelization.

File: calc/HamiltonianTwo.cpp (hunk 2 above)
Comment (seweber) [inline calc/HamiltonianTwo.cpp]: Again, lacking support of Visual Studio 2015.

File: calc/HamiltonianTwo.cpp (hunk 3 above)
Comment (seweber) [inline calc/HamiltonianTwo.cpp]: Again, lacking support of Visual Studio 2015.

File: calc/HamiltonianTwo.cpp (hunk 5 above)
Comment (seweber) [inline calc/HamiltonianTwo.cpp]: Again, lacking support of Visual Studio 2015.

File: gui/CMakeLists.txt (hunk 7 above)
Comment (seweber) [inline gui/CMakeLists.txt]: Oh, yes, I fixed it.

**Comment by seweber** [comment]
I would keep the openSUSE cross-platform build until we created the Nullsoft installer on AppVeyor. Up to now, we just create a zip file containing all the components of the pairinteraction software.

File: calc/HamiltonianOne.cpp (hunk 1 above)
Comment (seweber) [inline calc/HamiltonianOne.cpp]: Ok, I will cast the variable outside.

**Pull request merged**